app/controllers/api/v0/user_groups_controller.rb

```diff
@@ -19,10 +19,11 @@ class Api::V0::UserGroupsController < Api::V0::ApiController
   end
 
   # Filters the list of groups based on given parameters.
-  private def filter_groups_for_parameters(groups: [], is_active: nil)
+  private def filter_groups_for_parameters(groups: [], is_active: nil, parent_group_id: nil)
     groups.reject do |group|
       (
-        !is_active.nil? && is_active != group.is_active
+        (!is_active.nil? && is_active != group.is_active) ||
+        (parent_group_id.present? && group.parent_group_id != parent_group_id)
```

Comment:
Inconsistency between using `!foo.nil?` and `foo.present?` (which I believe both end up doing the same thing)

Reply:
Ah, I remember we had a discussion about this before.
Please add a comment about this intended inconsistency (because of false vs nil) to make sure that future me doesn't complain about this in the next few code reviews :stuck_out_tongue:

Reply:
Done